controllers/request_controller.go

```diff
@@ -297,22 +297,43 @@ func (r *RequestController) reconcileStatusPatch(
 	}
 
 	// Check if we have still time to requeue & retry
-	isPendingError := errors.As(err, &signer.PendingError{})
+	isPending := errors.As(err, &signer.PendingError{})
 	isPermanentError := errors.As(err, &signer.PermanentError{})
 	pastMaxRetryDuration := r.Clock.Now().After(requestObject.GetCreationTimestamp().Add(r.MaxRetryDuration))
-	if !isPendingError && (isPermanentError || pastMaxRetryDuration) {
-		// fail permanently
+	switch {
+	case isPending:
+		// Signing is pending, wait more.
+		//
+		// The PendingError has a misleading name: although it is an error,
+		// it isn't an error. It just means that we should poll again later.
+		// Its message gives the reason why the signing process is still in
+		// progress. Thus, we don't log any error.
+		logger.V(1).WithValues("reason", err.Error()).Info("Signing in progress.")
+		statusPatch.SetPending(fmt.Sprintf("Signing still in progress. Reason: %s", err))
+
+		// Let's not trigger an unnecessary reconciliation when we know that the
+		// user-defined condition was changed and will trigger a reconciliation.
+		if didCustomConditionTransition {
+			return result, statusPatch, reconcile.TerminalError(err) // apply patch, done
+		} else {
+			return result, statusPatch, err // apply patch, requeue with backoff
```

Comment:
Same here, set result.requeue=true and not return an error?

Reply:
Correct, I definitely don't want to return an error here.

Reply:
This commit fixes this: https://github.com/cert-manager/issuer-lib/pull/59/commits/ec382adc8454a19f77f2837f4961df35c302d16d